### DISCUSSION TITLE
Major Release. pyuvm 3.0.0 with UVM RAL support

### DISCUSSION THREAD

**Discussion by raysalemi** [pyuvm/pyuvm#220] [Announcements]
# Initial RAL Support

**pyuvm**, as I originally wrote it, did not support the UVM Register Access Layer. Fortunately, our outstanding contributors, especially @EngRaff92 and @crteja, took a leadership role in bringing RAL to pyuvm. 

pyuvm* 3.0.0 is the first RAL release. This initial release supports most of UVM RAL. We need to add backdoor support and other features, but we have started. The release includes a TinyALU example using RAL.

Thank you to everyone who contributed to this major release, pyuvm 3.0.0!

# Sphinx API Documentation

**pyuvm** now has Sphinx-generated documentation and updated docstrings that provide an API reference guide. You can read the documentation on [GitHub Pages](https://pyuvm.github.io/pyuvm/)

<hr /><em>This discussion was created from the release <a href='https://github.com/pyuvm/pyuvm/releases/tag/3.0.0'>Major Release. pyuvm 3.0.0 with UVM RAL support</a>.</em>

**Comment by sibeov** [comment]
Thanks to all contributors! Appreciate it and I am grateful as a developer for the time and energy you all invest!

**Comment by bstei1010** [comment]
Thank you everyone for this release! Does anyone know of a tool for generating a PYUVM RAL model off SystemRDL inputs? I'm wondering if I need to create my own PeakRDL plugin.

**Reply by raysalemi** [reply]
> Does anyone know of a tool for generating a PYUVM RAL model off SystemRDL inputs? I'm wondering if I need 
> 
> 
> 
> 

I think you do need to create one.

**Reply by EngRaff92** [reply]
I believe I have something in place using the peakRDL flow but did not know we could easily integrate as plugin. The peakRDL did not work properly with memory as well. Will try to share but cannot be pushed in this repo anyway regards

**Reply by bstei1010** [reply]
I wrote a PeakRDL-PyUVM plugin today that appears to work pretty well. I just ran into some PyUVM issues with using multiple levels of reg_block hierarchy.  I can try testing it with a flattened hierarchy later.
Would be happy to share when it's done

**Reply by EngRaff92** [reply]
Ahahhah cool sorry I should publicly shared anyway looks like probably yours is better then mine. If you could share something I will try out at my end. Sorry if you had to work on while there were some work done already around but I did not think there was a need for

**Reply by bstei1010** [reply]
No problem at all. I still have a couple kinks to work out on my end. I'll share with you once it's finished
